hw/ip/prim/util/primgen.py

```diff
@@ -84,29 +95,72 @@ def _top_module_file(core_files, module_name):
             return file
 
 
-def _parse_module_header(generic_impl_filepath, module_name):
-    """ Parse a SystemVerilog file to extract the 'module' header
+def _parse_module_header_verible(generic_impl_filepath, module_name):
+    """ Parse a SystemVerilog file to extract the 'module' header using Verible
 
-    Return a dict with the following entries:
-    - module_header: the whole module header (including the 'module' keyword)
-    - package_import_declaration: import declarations
-    - parameter_port_list: parameter/localparam declarations in the header
-    - ports: the list of ports. The portlist can be ANSI or non-ANSI style (with
-      or without signal declarations; see the SV spec for details).
+    Implementation of _parse_module_header() which uses verible-verilog-syntax
+    to do the parsing. This is the primary implementation and is used when
+    supported Verible version is available.
+
+    See _parse_module_header() for API details.
     """
 
-    # TODO: Evaluate different Python SV parsers and use one instead of this
-    # beautiful regex. A more complete parser would ignore comments, for
-    # example, or properly handle matching parentheses ... Or write a minimal
-    # parser that can parse the module header.
-    #
-    # Initial evaluations:
-    # - https://github.com/PyHDI/Pyverilog:
-    #   requires Icarus and Python 3.6
-    # - https://kevinpt.github.io/hdlparse/:
-    #   Didn't do anything in my tests.
-    # - https://github.com/sepandhaghighi/verilogparser:
-    #   Doesn't support SystemVerilog
+    from verible_verilog_syntax import VeribleVerilogSyntax, PreOrderTreeIterator
+
+    verible = os.environ.get("VERIBLE_VERILOG_SYNTAX_BINARY",
+                             "verible-verilog-syntax")
+    parser = VeribleVerilogSyntax(executable=verible)
+
+    data = parser.parse_file(generic_impl_filepath, options={"skip_null": True})
+    if not data.tree:
+        raise ValueError("Unable to parse %s." %
+                         (generic_impl_filepath, ))
```

Comment:
Done

> And it would be much better API if the parse_file() method would throw an parse exception itself. Maybe you can follow-up upstream for that?

The object returned by `parse_file` contains list of errors reported by Verible. They could be returned in an exception object, but:
* There is a small chance that Verible recovered from errors and a tree has been generated.
* There are errors, the tree is not generated, but tokens list is (when requested)


As for list of errors, they are printed now:
```
Inspecting generic module /home/mglb/projects/systemverilog/src/opentitan/hw/ip/prim_generic/rtl/prim_generic_xor2.sv
Verible: parse error in line 12 column 0.
Verible: parse error in line 14 column 2.
Verible: parse error in line 14 column 31.
Unable to parse '/home/mglb/projects/systemverilog/src/opentitan/hw/ip/prim_generic/rtl/prim_generic_xor2.sv'.
(...)
```